Gems/RecastNavigation/Code/Source/Misc/RecastNavigationMeshCommon.cpp

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <DetourNavMeshBuilder.h>
+#include <AzCore/Console/Console.h>
+#include <AzCore/Debug/Profiler.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+#include <Misc/RecastNavigationMeshCommon.h>
+#include <RecastNavigation/RecastNavigationPhysXProviderBus.h>
+
+AZ_DEFINE_BUDGET(Navigation);
+
+namespace RecastNavigation
+{
+    NavigationTileData RecastNavigationMeshCommon::CreateNavigationTile(TileGeometry* geom,
```

Comment:
suggestion: Consider breaking some of the steps out into separate functions (could have private static functions in the .cpp that just take parameters to modify, this keeps the public interface clean but still helps break things up)

Reply:
Done!